packages/api/models/grantCycle.js

```diff
@@ -67,6 +67,40 @@ module.exports = (sequelize, DataTypes) => {
               throw new Error('something went wrong on validation');
             }
           }
+
+          try {
+            const result = await GrantCycle.findAll({
+              where: {
+                [Op.and]: [{
+                  id: {
+                    [Op.not]: [instance.id]
+                  }}, {
+                  [Op.or]: [{
+                    openedOn: {
+                      [Op.between]: [instance.openedOn, instance.closedOn]
+                    }
+                  }, {
+                    closedOn: {
+                      [Op.between]: [instance.openedOn, instance.closedOn]
+                    }
+                  }, {
+                    [Op.and]: [{
+                      openedOn: {[Op.lt]: instance.openedOn},
+                      closedOn: {[Op.gt]: instance.closedOn}
+                    }]
+                  }]
+                }]
+              }
+            });
+            if (result.length) {
+              throw new ValidationError("Dates overlap with existing grant cycle");
+            }
+          }
+          catch (error) {
```

Comment:
Seems like we don't really care about the type of error, since if there are any existing overlapping GrantCycles, the ValidationError is thrown. Made a general catch all.